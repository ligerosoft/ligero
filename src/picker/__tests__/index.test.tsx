import { mount } from 'enzyme';
import React from 'react';
import Picker from '..';

const simpleColumns = ['red', 'blue', 'green'];
describe('Picker', () => {
  it('it should basic render corretly', () => {
    const wrapper = mount(<Picker columns={simpleColumns} />);
    expect(wrapper).toMatchSnapshot();
  });

  it('response events', async () => {
    const onChange = jest.fn();
    const onCancel = jest.fn();
    const onConfirm = jest.fn();
    const wrapper = mount(
      <Picker
        columns={simpleColumns}
        onCancel={onCancel}
        onChange={onChange}
        onConfirm={onConfirm}
      />,
    );
    wrapper.find('.ligero-picker__cancel').simulate('click');
    expect(onCancel).toHaveBeenCalled();

    wrapper.find('.ligero-picker__confirm').simulate('click');
    expect(onConfirm).toHaveBeenCalled();
  });
  it('set default index', () => {
    const wrapper = mount(<Picker columns={simpleColumns} />);
    expect(wrapper).toMatchSnapshot();

    wrapper.setProps({ defaultIndex: 2 });
    expect(wrapper).toMatchSnapshot();
  });
});
