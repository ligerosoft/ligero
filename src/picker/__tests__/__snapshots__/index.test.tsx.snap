// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Picker it should basic render corretly 1`] = `
<@ligero/picker
  columns={
    Array [
      "red",
      "blue",
      "green",
    ]
  }
>
  <div
    className="ligero-picker"
  >
    <div
      className="ligero-picker__toolbar"
    >
      <button
        className="ligero-picker__cancel"
        onClick={[Function]}
        type="button"
      >
        取消
      </button>
      <div
        className="ligero-picker__title"
      />
      <button
        className="ligero-picker__confirm"
        onClick={[Function]}
        type="button"
      >
        确定
      </button>
    </div>
    <div
      className="ligero-picker__columns"
      style={
        Object {
          "height": 240,
        }
      }
    >
      <PickerColumn
        columns={
          Array [
            "red",
            "blue",
            "green",
          ]
        }
        onChange={[Function]}
      >
        <div
          className="ligero-picker-column"
          onTouchCancel={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          onTouchStart={[Function]}
        >
          <div
            className="ligero-picker-column__wrapper"
            onTransitionEnd={[Function]}
            style={
              Object {
                "transform": "translate3d(0, 96px, 0)",
                "transitionDuration": "0ms",
                "transitionProperty": "none",
              }
            }
          >
            <div
              className="ligero-picker-column__item"
              key="red"
              onClick={[Function]}
              style={
                Object {
                  "height": 48,
                }
              }
            >
              red
            </div>
            <div
              className="ligero-picker-column__item"
              key="blue"
              onClick={[Function]}
              style={
                Object {
                  "height": 48,
                }
              }
            >
              blue
            </div>
            <div
              className="ligero-picker-column__item"
              key="green"
              onClick={[Function]}
              style={
                Object {
                  "height": 48,
                }
              }
            >
              green
            </div>
          </div>
        </div>
      </PickerColumn>
      <div
        className="ligero-picker__mask"
        style={
          Object {
            "backgroundSize": "100% 96px",
          }
        }
      />
      <div
        className="ligero-hairline-unset--top-bottom ligero-picker__hairline"
        style={
          Object {
            "height": 48,
          }
        }
      />
    </div>
  </div>
</@ligero/picker>
`;

exports[`Picker set default index 1`] = `
<@ligero/picker
  columns={
    Array [
      "red",
      "blue",
      "green",
    ]
  }
>
  <div
    className="ligero-picker"
  >
    <div
      className="ligero-picker__toolbar"
    >
      <button
        className="ligero-picker__cancel"
        onClick={[Function]}
        type="button"
      >
        取消
      </button>
      <div
        className="ligero-picker__title"
      />
      <button
        className="ligero-picker__confirm"
        onClick={[Function]}
        type="button"
      >
        确定
      </button>
    </div>
    <div
      className="ligero-picker__columns"
      style={
        Object {
          "height": 240,
        }
      }
    >
      <PickerColumn
        columns={
          Array [
            "red",
            "blue",
            "green",
          ]
        }
        onChange={[Function]}
      >
        <div
          className="ligero-picker-column"
          onTouchCancel={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          onTouchStart={[Function]}
        >
          <div
            className="ligero-picker-column__wrapper"
            onTransitionEnd={[Function]}
            style={
              Object {
                "transform": "translate3d(0, 96px, 0)",
                "transitionDuration": "0ms",
                "transitionProperty": "none",
              }
            }
          >
            <div
              className="ligero-picker-column__item"
              key="red"
              onClick={[Function]}
              style={
                Object {
                  "height": 48,
                }
              }
            >
              red
            </div>
            <div
              className="ligero-picker-column__item"
              key="blue"
              onClick={[Function]}
              style={
                Object {
                  "height": 48,
                }
              }
            >
              blue
            </div>
            <div
              className="ligero-picker-column__item"
              key="green"
              onClick={[Function]}
              style={
                Object {
                  "height": 48,
                }
              }
            >
              green
            </div>
          </div>
        </div>
      </PickerColumn>
      <div
        className="ligero-picker__mask"
        style={
          Object {
            "backgroundSize": "100% 96px",
          }
        }
      />
      <div
        className="ligero-hairline-unset--top-bottom ligero-picker__hairline"
        style={
          Object {
            "height": 48,
          }
        }
      />
    </div>
  </div>
</@ligero/picker>
`;

exports[`Picker set default index 2`] = `
<@ligero/picker
  columns={
    Array [
      "red",
      "blue",
      "green",
    ]
  }
  defaultIndex={2}
>
  <div
    className="ligero-picker"
  >
    <div
      className="ligero-picker__toolbar"
    >
      <button
        className="ligero-picker__cancel"
        onClick={[Function]}
        type="button"
      >
        取消
      </button>
      <div
        className="ligero-picker__title"
      />
      <button
        className="ligero-picker__confirm"
        onClick={[Function]}
        type="button"
      >
        确定
      </button>
    </div>
    <div
      className="ligero-picker__columns"
      style={
        Object {
          "height": 240,
        }
      }
    >
      <PickerColumn
        columns={
          Array [
            "red",
            "blue",
            "green",
          ]
        }
        defaultIndex={2}
        onChange={[Function]}
      >
        <div
          className="ligero-picker-column"
          onTouchCancel={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          onTouchStart={[Function]}
        >
          <div
            className="ligero-picker-column__wrapper"
            onTransitionEnd={[Function]}
            style={
              Object {
                "transform": "translate3d(0, 96px, 0)",
                "transitionDuration": "0ms",
                "transitionProperty": "none",
              }
            }
          >
            <div
              className="ligero-picker-column__item"
              key="red"
              onClick={[Function]}
              style={
                Object {
                  "height": 48,
                }
              }
            >
              red
            </div>
            <div
              className="ligero-picker-column__item"
              key="blue"
              onClick={[Function]}
              style={
                Object {
                  "height": 48,
                }
              }
            >
              blue
            </div>
            <div
              className="ligero-picker-column__item"
              key="green"
              onClick={[Function]}
              style={
                Object {
                  "height": 48,
                }
              }
            >
              green
            </div>
          </div>
        </div>
      </PickerColumn>
      <div
        className="ligero-picker__mask"
        style={
          Object {
            "backgroundSize": "100% 96px",
          }
        }
      />
      <div
        className="ligero-hairline-unset--top-bottom ligero-picker__hairline"
        style={
          Object {
            "height": 48,
          }
        }
      />
    </div>
  </div>
</@ligero/picker>
`;
